client/components/Games/SnakeGameBoard.tsx

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable jsx-a11y/no-noninteractive-tabindex */
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+import { useEffect, useRef, useState } from 'react'
+import { nextPosition, randomCoords } from '../../utils/snakeUtils'
+import Button from '../UI/Button'
+
+interface Props {
+  gridSize: number
+  speed: number
+  numOfObstacles: number
+}
+
+export default function SnakeGameBoard({
+  gridSize,
+  speed,
+  numOfObstacles,
+}: Props) {
+  const nums = new Array(gridSize).fill(0).map((_, i) => i) // used to generate the table programatically
+
+  const tableRef = useRef<HTMLInputElement>(null) // used in auto focusing on the table
+  const initialCoords = useRef(randomCoords(gridSize, numOfObstacles + 1))
+  const [initialPosition, ...obstacles] = initialCoords.current
+
+  const [head, setHead] = useState(initialPosition) // where the snake's head is. Movement in the grid is based on this
+  const [snake, setSnake] = useState([initialPosition]) // every segment of the snake
+  const [food, setFood] = useState(
+    randomCoords(gridSize, 1, [...snake, ...obstacles])[0],
+  )
+  const [score, setScore] = useState(0)
+  const [direction, setDirection] = useState('none')
+  const [gameState, setGameState] = useState('alive')
+
+  useEffect(() => {
+    if (tableRef.current) {
+      tableRef.current.focus()
+    }
+
+    const intervalId = setInterval(() => {
+      if (head) {
+        const next = nextPosition(head, direction, gridSize, setGameState) // nextPosition also checks if the snake will go out of bounds
+        const [_noggin, ...tail] = snake
+        if (next != null && [...tail, ...obstacles].includes(next)) {
+          // check if snake crashes into obstacle or itself, but can't crash into it's own head
+          setGameState('dead')
+        } else if (next && food === next) {
+          //snake gets longer if it eats food (expands into new square, and doesn't short the tail)
+          setHead(next)
+          setSnake([next, ...snake])
+          setFood(randomCoords(gridSize, 1, [...snake, ...obstacles])[0])
+          setScore(score + 1)
+        } else if (next) {
+          // doesn't collide with anything, moves into new square and removes last quare of tail
```

Comment:
just a typo in the comment, 'quare' instead of square

Reply:
fixed